components/admin/lessons/AdminModuleInputs.tsx

```diff
@@ -0,0 +1,185 @@
+import React, { useEffect, useState } from 'react'
+import {
+  AddModuleMutation,
+  UpdateModuleMutation,
+  useAddModuleMutation,
+  useUpdateModuleMutation
+} from '../../../graphql'
+import { formChange } from '../../../helpers/formChange'
+import { FormCard, MD_INPUT, TextField } from '../../FormCard'
+import { AlertFillIcon, CheckCircleIcon } from '@primer/octicons-react'
+import styles from '../../../scss/adminModuleInputs.module.scss'
+import { Spinner } from 'react-bootstrap'
+import { get } from 'lodash'
+import { RefetchQueriesFunction, ApolloError } from '@apollo/client'
+
+type Module = { id: number; name: string; content: string }
+
+type Props = {
+  lessonId: number
+  refetch: RefetchQueriesFunction
+  title?: string
+  onAddModule?: (
+    m:
+      | (AddModuleMutation['addModule'] & { lesson: { id: number } })
+      | (UpdateModuleMutation['updateModule'] & { lesson: { id: number } })
+      | null,
+    e: { name: string; content: string } | null
+  ) => void
+  module?: Module | undefined
+}
+
+enum Error {
+  InvalidData = 'missing module name or description'
+}
+
+const initValues = (
+  nameValue?: string,
+  descValue?: string
+): [TextField, TextField] => [
+  {
+    title: 'Module Name',
+    value: nameValue || ''
+  },
+  {
+    title: 'Description',
+    type: MD_INPUT,
+    value: descValue || ''
+  }
+]
+
+const AdminModuleInputs = ({
+  title,
+  lessonId,
+  onAddModule,
+  module,
+  refetch
+}: Props) => {
+  const [formOptions, setFormOptions] = useState<any>(initValues())
+  const [name, content] = formOptions
+
+  useEffect(
+    () => setFormOptions(initValues(module?.name, module?.content)),
+    [module]
+  )
+
+  const [moduleMutation, { data, error, loading }] = module
+    ? useUpdateModuleMutation({
+        variables: {
+          id: module.id,
+          content: content.value,
+          lessonId,
+          name: name.value
+        }
+      })
+    : useAddModuleMutation({
+        variables: {
+          content: content.value,
```

Comment:
lines 76 - 80 are repeated on lines 68 - 73. Can you reduce the code duplication?